src/test/java/bio/terra/common/iam/AuthenticatedUserRequestTest.java

```diff
@@ -0,0 +1,75 @@
+package bio.terra.common.iam;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import bio.terra.common.exception.ApiException;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.UUID;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Tag("unit")
+public class AuthenticatedUserRequestTest {
+
+  private static final Logger logger = LoggerFactory.getLogger(AuthenticatedUserRequestTest.class);
+
+  private static final String EmailAddress = "test@example.com";
+  private static final UUID RequestId = UUID.randomUUID();
+  private static final String SubjectId = "Subject";
+  private static final String Token = "0123.456-789AbCd";
+
+  public void validateJsonSerialization(AuthenticatedUserRequest request)
+      throws JsonProcessingException {
+    ObjectMapper objectMapper = new ObjectMapper();
+    String str = objectMapper.writeValueAsString(request);
+    logger.debug(String.format("Serialized AuthenticatedUserRequest: '%s'", str));
+
+    AuthenticatedUserRequest deserialized =
+        objectMapper.readValue(str, AuthenticatedUserRequest.class);
+    assertEquals(request, deserialized);
+    assertEquals(request.hashCode(), deserialized.hashCode());
+  }
+
+  @Test
+  public void fullBuilder() throws Exception {
+    AuthenticatedUserRequest req =
+        new AuthenticatedUserRequest.Builder()
+            .setEmail(EmailAddress)
+            .setReqId(RequestId)
+            .setSubjectId(SubjectId)
+            .setToken(Token)
+            .build();
+
+    assertEquals(EmailAddress, req.getEmail());
+    assertEquals(SubjectId, req.getSubjectId());
+    assertEquals(Token, req.getToken());
+    assertEquals(Token, req.getRequiredToken());
+    assertNotNull(req.getReqId());
+    validateJsonSerialization(req);
+  }
+
+  @Test
+  public void emptyBuilder() throws Exception {
+    AuthenticatedUserRequest req = new AuthenticatedUserRequest.Builder().build();
+    assertNull(req.getEmail());
+    assertNull(req.getSubjectId());
+    assertNull(req.getToken());
+    assertNotNull(req.getReqId());
+    assertThrows(ApiException.class, req::getRequiredToken);
+    validateJsonSerialization(req);
+  }
+
+  @Test
+  public void equality() {
+    AuthenticatedUserRequest req = new AuthenticatedUserRequest.Builder().build();
+    assertEquals(req, req);
```

Comment:
probably more interesting to assert equality for a full build.

Reply:
Yes, this test was lacking in substance and documentation, I improved it in latest push.